File: Day 01/Java/Java/BinarySearch.java
package Java;

/**
 * Regular binary search
 */
public class BinarySearch {

    public static int BinarySearch( int input[], int search) {
        int low = 0;
        int high = input.length - 1;
        int mid;
        while (low <= high) {
            mid = low + ((high - low) / 2);
            if (input[mid] == search) {
                return mid;
            } else if (input[mid] < search) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return -1;
    }

    public static void main(String args[]) {
        final int arr1[] = {1, 2, 4, 5, 7, 8};
        System.out.println(BinarySearch(arr1, -1));
        System.out.println(BinarySearch(arr1, 1));
        System.out.println(BinarySearch(arr1, 8));
        System.out.println(BinarySearch(arr1, 2));
    }
}

File: Day 01/Python/Python/BinarySearch.py
def binary_search(l, num_find):
    start = 0
    end = len(l) - 1
    mid = (start + end) // 2
    
    found = False
    position = -1

    while start <= end:
        if l[mid] == num_find:
            found = True
            position = mid
            break
        
        if num_find > l[mid]:
            start = mid + 1
            mid = (start + end) // 2
        else:
            end = mid - 1
            mid = (start + end) // 2

    return (found, position)

if __name__=='__main__':
    l = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    num = input("Enter a number:")
    num=int(num)
    found = binary_search(l, num)
    if found[0]:
        print('Number %d found at position %d'%(num, found[1]+1))
    else:
        print('Number %d not found'%num)